test/Api.IntegrationTest/Controllers/SecretsControllerTest.cs

```diff
@@ -58,7 +58,7 @@ public async Task CreateSecret()
         var result = await response.Content.ReadFromJsonAsync<Secret>();
 
         Assert.NotNull(result);
-        Assert.Equal(request.Key, result.Key);
+        Assert.Equal(request.Key, result!.Key);
```

Comment:
If we update to a newer version of Xunit, 2.4.2 or greater, we won't need the `result!.Key` because the `Assert.NotNull(result);` will work with C# nullable types.

Reply:
Let's do that outside the SM branch.